### PR TITLE
Add CodeSignVerification to signed yaml build and fixed dll signing

### PR DESCRIPTION
**Pull request by WilliamXieMSFT**

### CODE REVIEW

File: build/yaml/steps/microbuild/codesignverify.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Step Template: MicroBuild Signing and Verification
+
+parameters:
+    SignType: None
+
+steps:
+
+- ${{ if notIn(parameters.SignType, '', 'None') }}:
```
Comment (delmyers) [inline build/yaml/steps/microbuild/codesignverify.yaml]: >if notIn(parameters.SignType, '', 'None') [](start = 6, length = 41)

Consider doing the positive approach, if the parameter is set to "Real" or "Test", then set the variable.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,8 @@ Please format the changes as follows:
   + Add configuration for InstrumentationMethod loglevels
 + Updates:
   + Added documentation around the logging infrastructure
++ BugFixes:
+  + Fix codesigning ProfilerProxy
```
Comment (delmyers) [inline CHANGELOG.md]: >Fix [](start = 4, length = 3)

Is this really what you did? Seems like what you did was add code sign verification.

Comment (WilliamXieMSFT) [reply CHANGELOG.md]: Yep, it turned out the typo in the ProfilerProxy project silently failed and did not sign the assembly.

---
In reply to: [356697030](https://github.com/microsoft/CLRInstrumentationEngine/pull/181#discussion_r356697030) [](ancestors = 356697030)

File: build/yaml/steps/microbuild/codesignverify.yaml (hunk 1 above)
Comment (WilliamXieMSFT) [reply build/yaml/steps/microbuild/codesignverify.yaml]: I wasn't able to get it working earlier, I'll try this again

---
In reply to: [356695467](https://github.com/microsoft/CLRInstrumentationEngine/pull/181#discussion_r356695467) [](ancestors = 356695467)

Comment (WilliamXieMSFT) [reply build/yaml/steps/microbuild/codesignverify.yaml]: Turns out we're pulling in a build variable which happens after the yaml generation.

---
In reply to: [356724316](https://github.com/microsoft/CLRInstrumentationEngine/pull/181#discussion_r356724316) [](ancestors = 356724316,356695467)

**Comment by WilliamXieMSFT** [comment]
>     <FileVersion>$(ProxyFileVersionMajor),$(ProxyFileVersionMinor),$(ProxyFileVersionPatch),0</FileVersion>

I might need to version bump this

---
Refers to: src/InstrumentationEngine.ProfilerProxy/InstrumentationEngine.ProfilerProxy.vcxproj:32 in e25d3f2. [](commit_id = e25d3f26a482647669be03195416db6f0957f3f6, deletion_comment = False)

**Pull request merged**